swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1281 @@
+
+    displacement_threshold: float, optional      
+        iteration halt criterion: Don't stop until every particle is displaced 
+        by distance < threshold * mean interparticle distance
+
+    delta_init: float or None, optional
+        initial normalization constant for particle motion in units of mean 
+        interparticle distance. If ``None`` (default), delta_init will be set 
+        such that the maximal displacement found in the first iteration is 
+        normalized to 1 mean interparticle distance.
+
+    delta_reduction_factor: float, optional
+        multiply the normalization constant for particle motion by this factor 
+        after every iteration. In certain difficult cases this might help the 
+        generation to converge if set to < 1.
+    
+    delta_min: float, optional
+        minimal normalization constant for particle motion in units of mean 
+        interparticle distance.
+   
+    redistribute_frequency: int, optional 
+        redistribute a handful of particles every ``redistribute_frequency`` 
+        iteration. How many particles are redistributed is controlled with the 
+        ``redistribute_fraction`` parameter.
+
+    redistribute_fraction: float, optional
+        fraction of particles to be redistributed when doing so. 
+
+    redistribute_fraction_reduction: float, optional
+        multiply the ``redistribute_fraction`` parameter by this factor every 
+        time particles are being redistributed. In certain difficult cases this
+        might help the generation to converge if set to < 1.
+    
+    no_redistribution_after: int, optional     
+        don't redistribute particles after this iteration.
+ 
+    intermediate_dump_frequency: int, optional
+        frequency of dumps of the current state of the iteration. If set to zero,
+        no intermediate results will be stored.
+
+    random_seed: int, optional
+        set a specific random seed
+
+
+
+    Returns
+    ------------
+
+    icRunParams: dict
+        dict containing the parameters stored in a way such that 
+        ``generate_IC_for_given_density()`` understands them.
+
+
+    Notes
+    -----------
+    
+    + The returned dict is a required argument to call 
+        ``generate_IC_for_given_density()``
+    """
+
+    icRunParams = {}
+
+    icRunParams["ITER_MAX"] = iter_max
+    icRunParams["CONVERGE_THRESH"] = convergence_threshold
+    icRunParams["TOLERANCE_PART"] = tolerance_part
+    icRunParams["DISPL_THRESH"] = displacement_threshold
+    icRunParams["DELTA_INIT"] = delta_init
+    icRunParams["DELTA_REDUCT"] = delta_reduction_factor
+    icRunParams["DELTA_MIN"] = delta_min
+    icRunParams["REDIST_FREQ"] = redistribute_frequency
+    icRunParams["REDIST_FRAC"] = redistribute_fraction
+    icRunParams["REDIST_STOP"] = no_redistribution_after
+    icRunParams["REDIST_REDUCT"] = redistribute_fraction_reduction
+    icRunParams["DUMPFREQ"] = intermediate_dump_frequency
+
+    global seed_set
```

Comment:
In case somebody (like me) has the very bright idea to call the setter routine twice. In that case, the seed gets reset, and that's most likely not something you want. Example:

```
>>> import numpy as np
>>> np.random.seed(666)
>>> np.random.uniform(size=10)
array([0.70043712, 0.84418664, 0.67651434, 0.72785806, 0.95145796,
       0.0127032 , 0.4135877 , 0.04881279, 0.09992856, 0.50806631])
>>> np.random.seed(666)
>>> np.random.uniform(size=10)
array([0.70043712, 0.84418664, 0.67651434, 0.72785806, 0.95145796,
       0.0127032 , 0.4135877 , 0.04881279, 0.09992856, 0.50806631])

```

Reply:
Has been removed in the updated version.